packages/aws-arch/src/internal/mapping/helpers.ts

```
@@ -0,0 +1,160 @@
+  resourceName: string;
+}
+
+/** Parse CloudFormation resource type string into parts */
+export function parseCfnType(
+  cfnType: CfnSpec.ResourceType
+): CfnResourceTypeParts {
+  return cfnType.match(
+    /(?<provider>\w+)::(?<serviceName>\w+)::(?<resourceName>\w+)/
+  )!.groups as any;
+}
+
+/** Definitaion of asset mappings to CloudFormation type */
+interface CfnTypeAssets {
+  readonly serviceName?: AwsAsset.Service;
+  readonly resourceName?: AwsAsset.Resource;
+  readonly generalIcon?: AwsAsset.GeneralIcon;
+}
+
+// VPC is nested in EC2 within CloudFormation but separate in other systems
+const VpcAssetComparables: Record<string, string> = Object.fromEntries(
+  Object.keys(AwsAsset.Resources).reduce((entries, key) => {
+    if (key.startsWith("vpc_")) {
+      entries.push([normalizeComparisonString(key.replace("vpc_", "")), key]);
+    }
+    return entries;
+  }, [] as [string, string][])
+);
+
+/** Find asset type details for given CloudFormation type */
+export function findCfnTypeAssets(
+  cfnType: CfnSpec.ResourceType
+): CfnTypeAssets {
+  const cfn = parseCfnType(cfnType);
+
+  let serviceName: AwsAsset.Service | undefined;
+  let resourceName: AwsAsset.Resource | undefined;
+
+  // handle edge cases (eg: EC2::VPCxxx is vpc:xxx in assets)
+  if (cfn.serviceName === "EC2") {
+    if (cfn.resourceName.startsWith("VPC")) {
+      serviceName = "vpc";
+      cfn.resourceName = cfn.resourceName.replace(/^VPC/, "");
+    } else if (
+      normalizeComparisonString(cfn.resourceName) in VpcAssetComparables
+    ) {
+      serviceName = "vpc";
+      cfn.resourceName = normalizeComparisonString(cfn.resourceName);
+    }
+  }
+
+  if (serviceName == null) {
+    try {
+      serviceName = resolveServiceName(cfn.serviceName);
+    } catch (e) {
+      console.warn((e as Error).message, cfnType);
+    }
+  }
+
+  if (resourceName == null) {
+    try {
+      resourceName = resolveResourceName(cfn.resourceName, serviceName);
+    } catch {}
```

Do we need to catch and swallow the error? Why not just omit and have it bubble up?

---

I will add the follow comment to the code explaining reason to just ignore

> // There are lots of low-level cfn resource definitions without mappings to other systems,
> // for this reason we just ignore unresolved resources without spamming the console or
>  // bubbling the error as this is expected in large percent of cases.